src/NServiceBus.Core/DelayedDelivery/TimeoutManager/DispatchTimeoutBehavior.cs

```diff
@@ -31,7 +31,8 @@ public DispatchTimeoutBehavior(IDispatchMessages dispatcher, IPersistTimeouts pe
 
             var outgoingMessage = new OutgoingMessage(context.MessageId, timeoutData.Headers, timeoutData.State);
             var transportOperation = new TransportOperation(outgoingMessage, new UnicastAddressTag(timeoutData.Destination), dispatchConsistency);
-            await dispatcher.Dispatch(new TransportOperations(transportOperation), context.Context).ConfigureAwait(false);
+            var transportTransaction = context.Context.GetOrCreate<TransportTransaction>();
```

Comment:
If the intention is to keep `TransportTransaction` and the context bag separate, why would we want to store a newly created one in the context bag here?